src/teehr/visualization/dataframe_accessor.py

```diff
@@ -28,15 +32,60 @@ class TEEHRDataFrameAccessor:
     def __init__(self, pandas_obj):
         """Initialize the class."""
         self._df = pandas_obj
+        self._df.attrs = pandas_obj.attrs
         self._validate(pandas_obj)
 
     @staticmethod
     def _validate(obj):
-        """Validate the DataFrame object columns."""
-        if "location_id" not in obj.columns:
-            raise AttributeError("Must have 'location_id'.")
-        if obj.index.size == 0:
-            raise AttributeError("DataFrame must have data.")
+        """Validate the DataFrame object."""
+        if 'table_type' not in obj.attrs:
+            raise AttributeError("Not DataFrame Attribute 'table_type'"
+                                 " defined.")
+
+        if obj.attrs['table_type'] == 'timeseries':
+
+            # check for expected fields
+            fields_list = Timeseries.get_field_names()
+            missing = []
+            for field in fields_list:
+                if field not in obj.columns:
+                    missing.append(field)
+            if len(missing) != 0:
+                raise AttributeError(f'''
```

Comment:
we mostly use double quotes for strings.

also if you start your strings on a new line you can move the intend back to the left and avoid having the message split across so many lines.

Reply:
Attempted to reformat all line spacing and quote inconsistencies where applicable. Let me know if anything seems off!